dev-python/gmpy/files/gmpy-2.1.0_beta5-pyhash-nan.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/src/gmpy2_hash.c b/src/gmpy2_hash.c
```

Comment:
Is this an upstream patch? Please add a link or a comment where it comes from.

Reply:
There is a comment in the ebuild explaining the patch. The upstream doesn't work with python 3.10 because the constant was removed in 3.10. Should probably submit the patch upstream, but that'll only be in the next release.

Reply:
The point is that we generally prefer patches blessed upstream, or at least submitted somewhere official where we can figure out what's happening.

Reply:
I'll try submitting it upstream and see what they say.

Reply:
It also seems that the patch that I made that disables the failings tests have been resolved upstream already, at least so said in https://github.com/aleaxit/gmpy/issues/296 but it's not in a release yet.

Reply:
Find the upstream commit then.

Reply:
It was this commit: https://github.com/aleaxit/gmpy/commit/db7ce2ef46fab84e7b9c32b05725e9b02e8cf206. The patch did't apply cleanly so I had to manually copy-past from the commit, because too much changed in a ~year since the last release, for it to apply, plus some functions were renamed.

Reply:
Cool, I see you've backported more changes and made things work ;-).